### ISSUE TITLE
trim_punctuation fails on strings with non-ASCII

### ISSUE THREAD

**Issue by dunn** [traject/traject#133]
I haven't been able to generate the raw MARC needed for a test case, but in my Traject configuration I have

``` ruby
to_field 'alternative', extract_marc('740a', trim_punctuation: true)
```

and it's importing some MARCXML with

``` xml
  <datafield tag="740" ind1="3" ind2=" ">
    <subfield code="a">Le réve.</subfield>
  </datafield> 
```

The period is not stripped from the resulting string, but is when "Le réve" is replaced with "Le reve".

**Comment by jrochkind** [comment]
Can you confirm for sure it's an ascii period (byte 46) and not some other weird period-looking thing?

**Comment by dunn** [comment]
It is, yes.

**Comment by redlibrarian** [comment]
I think I've confirmed this behaviour and written a test for it.

**Comment by billdueber** [comment]
I have a theory -- I wonder if [this](https://github.com/traject/traject/blob/master/lib/traject/macros/marc21.rb) is messing up because it's not thinking the accented character is a `/\w/` character. Can you write a quick test and I'll test out various unicode-aware options?

**Comment by redlibrarian** [comment]
Yeah, I've done some digging and it does look like it's an encoding thing. Changing it to [[:word:]] trims the punctuation, but also removes the accent. I'll issue a PR with the failing test.

**Comment by billdueber** [comment]
Fixed by #135

**Comment by dunn** [comment]
Great, thanks all!
